hyphae/src/myceli_api.rs

```diff
@@ -0,0 +1,118 @@
+use anyhow::{bail, Result};
+use messages::{
+    ApplicationAPI, DataProtocol, Message, MessageChunker, SimpleChunker, TransmissionBlock,
+};
+use std::net::{ToSocketAddrs, UdpSocket};
+use std::rc::Rc;
+use std::thread::sleep;
+use std::time::Duration;
+use tracing::{debug, info, warn};
+
+const MTU: u16 = 60;
+
+pub struct MyceliApi {
+    address: String,
+    socket: Rc<UdpSocket>,
+    listen_address: String,
+}
+
+impl MyceliApi {
+    pub fn new(address: &str) -> Self {
+        let socket = Rc::new(UdpSocket::bind("127.0.0.1:0").unwrap());
+        socket
+            .set_read_timeout(Some(Duration::from_millis(500)))
+            .unwrap();
```

Comment:
`unwrap` here, might be OK since it's at startup.  If it's OK, might be good to wrap in an `expect` instead.